dia_tap_xero/credentials.py

```diff
@@ -62,33 +43,18 @@ def build_oauth_headers(config):
     return oauth_headers
 
 
-def _on_giveup(details):
-    _, body = details["args"]
-    LOGGER.error(
-        "Credentials could not be saved to S3. "
-        + "You will need to re-authorize the application."
-    )
-
-
-@backoff.on_exception(
-    backoff.expo, Exception, max_tries=5, on_giveup=_on_giveup, factor=2
-)
-def _upload(obj, body):
-    obj.put(Body=body.encode())
-
-
-def _write_to_s3(config):
-    creds = {x: config[x] for x in REFRESHABLE_KEYS}
-    _upload(_s3_obj(config), json.dumps(creds))
-
-
 def rotate_refresh_tokens(config, json_response):
 
     config["refresh_token"] = json_response["refresh_token"]
     config["access_token"] = json_response["access_token"]
 
-    # Rotate token on AWS
-    # TODO add functionality to rotate the token here
+    # Rotate refresh token on AWS SSM
+    SSM.put_parameter(
+        Name="/airflow/tap_xero_refresh_token",
+        Overwrite=True,
+        Value=json_response["refresh_token"],
+        Type="SecureString",
```

Comment:
doesn't the encryption/decryption require a KMS key? or it uses a default one?

Reply:
as long as you're in a valid AWS vault / identified that way, you don't need to pass anything else.